mmcomplex.raw.html

```diff
@@ -661,8 +661,7 @@
 A countable set may be infinite, but every element of the set can be
 mapped to a unique natural number.
 The natural numbers and the integers are countably infinite sets.
-Even the rational numbers are a countably infinite set; see
-<A HREF="qnnen.html">qnnen</A>.
+Even the rational numbers are a countably infinite set; see ~ qnnen .
```

Comment:
Using ~ here means that the page described here needs an extra resolution step from files by Metamath or relatives, and cannot be generated by some odd HTML tool any longer.  Why is that sensible?
If it is, there are still other links not converted, like cnpi for example.

Reply:
I believe the web site generation process is not doing such resolution currently, although metamath-exe has the capability to do it.

Reply:
The generation script does this for all files named `*.raw.html` here: https://github.com/metamath/metamath-website-scripts/blob/8d81a0c1ab3e433b335b4e6a3bdc364d277b4946/build-website.sh#L89 (I guess they also need to be copied here: https://github.com/metamath/metamath-website-scripts/blob/8d81a0c1ab3e433b335b4e6a3bdc364d277b4946/build-website.sh#L13 ). This mechanism has been in use quite extensively for mmil.raw.html for a long time and is also in use by mmset.raw.html .

As for why this is a good idea, one reason is to get the ability to typeset metamath notation, so that when editing the web pages we can just write `` ` a e. B ` `` and have it use the current markup from htmldef. Another reason is to check that references to theorems point to theorems which still exist, which is quite helpful given how much we rename things.

As for converting the rest of the file (`cnpi` for example), we'll get there eventually. As the discussion of `0.999...` elsewhere in this pull request shows, this is not a completely mechanical process (although it is mostly mechanical). But we have to start somewhere and the `ruc` text seemed like a good one because it is so out of date.

Reply:
Oh yes you are right. (The second link is not a mistake, the files are first copied into mpegif, then to mpeuni, then marked up)

Reply:
In addition, I've been using the https://github.com/metamath/set.mm/blob/develop/scripts/regen-from-raw script as I edit, to see what the formatted page looks like.

Reply:
If it was my page, I'd add a hint like `use regen-from-raw to transscript to pure html` somewhere at the beginning.

Reply:
I've submitted this at #3945